### PR TITLE
Fixes #7726 - Filtered out items no longer added when 'Select all' cl…

### PR DESCRIPTION
**Pull request by xprazak2**
…icked

### CODE REVIEW

File: app/assets/javascripts/jquery.multi-select.js [hunk 1]
```diff
@@ -72,7 +72,27 @@ function sanitize(value){
...
+        visibles = [],
+        selected = [];
+
+    $(formGroup).find('.ms-selectable').find('ul.ms-list').children().each( function () {
+      if(!$(this).hasClass('hide')) {
+        visibles.push($(this).first().text());
+      }
+    });
+
+    $(formGroup).find('select[multiple]').children().each( function () {
```
Comment (dLobatog) [inline app/assets/javascripts/jquery.multi-select.js]: Extract to a separate method

File: app/assets/javascripts/jquery.multi-select.js [hunk 2]
```diff
@@ -72,7 +72,27 @@ function sanitize(value){
 }
 
 $(document).on('click', '.ms-select-all', function () {
-    $(this).closest('.form-group').find('select[multiple]').multiSelect('select_all');
+  // can't use multiSelect('deselect_all') because it adds filtered out items too.
+    var formGroup = $(this).closest('.form-group'),
+        visibles = [],
+        selected = [];
+
+    $(formGroup).find('.ms-selectable').find('ul.ms-list').children().each( function () {
```
Comment (dLobatog) [inline app/assets/javascripts/jquery.multi-select.js]: Extract to a separate method

**Comment by dLobatog** [comment]
@xprazak2 Nice fix, I didn't even notice this bug! :+1: 

My only 2 comments - try to extract a few methods out of that js to make it easier to read - and write an integration test for this. If it's not tested - it's broken :smile:

File: app/assets/javascripts/jquery.multi-select.js [hunk 3]
```diff
@@ -72,9 +72,41 @@ function sanitize(value){
...
+
+    selected = itemsToSelect(formGroup);
+
+    $(formGroup).find('select[multiple]').multiSelect('select', selected);
     return false;
 });
+
+function findVisibleItems(formGroup) {
+  var visibles = [];
+  $(formGroup).find('.ms-selectable').find('ul.ms-list').children().each( function () {
```
Comment (tbrisker) [inline app/assets/javascripts/jquery.multi-select.js]: Why not something like `$(formGroup).find('.ms-selectable .ms-list :not(.hide)')` to find the visible items instead of this function and looping over every child? Let jQuery do what it does best - selecting stuff from the DOM.

File: app/assets/javascripts/host_edit.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+//= require host_edit_interfaces
```
Comment (tbrisker) [inline app/assets/javascripts/host_edit.js]: Is this related to this patch or just a leftover from something else? If it fixes a separate issue please put it in a different PR.

File: test/integration/organization_test.rb [hunk 5]
```diff
@@ -3,6 +3,8 @@
 class OrganizationIntegrationTest < ActionDispatch::IntegrationTest
   def setup
     FactoryGirl.create(:host)
+    Capybara.current_driver = Capybara.javascript_driver
```
Comment (dLobatog) [inline test/integration/organization_test.rb]: If you only need JS for this test, do this in the specific test - otherwise it'll slow down other tests in here (Poltergeist is slower than rack-test) and you don't need to login twice

**Comment by dLobatog** [comment]
@xprazak2 Tests are failing, please check that out.

**Comment by dLobatog** [comment]
Merged as 45be57cddfcc8d8895a96cd601c6ff132caacd8c, thanks @xprazak2!
